### PR TITLE
SDIT-1483 Add OffenderAlert entity and associated database changes

### PR DESCRIPTION
**Pull request by andymarke**
Introduced a new entity, OffenderAlert, to store offender alert data. Associated changes include adding new database table 'OFFENDER_ALERTS' and reference_codes for alerts. Also updated existing OffenderBooking entity and created a new repository: OffenderAlertRepository. Unit tests have been updated with a data builder for testing purposes.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderAlert.kt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  var modifyDatetime: LocalDateTime? = null
+
+  @Column(name = "CREATE_DATETIME", insertable = false, updatable = false)
+  lateinit var createDatetime: LocalDateTime
+
+  @Column(name = "CREATE_USER_ID", insertable = false, updatable = false)
+  @Generated
+  lateinit var createUserId: String
+
+  @Column(name = "AUDIT_TIMESTAMP", insertable = false, updatable = false)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderAlert.kt:109]: DPS are currently saying they want all of these, so mapping for now and no doubt will remove them once the realise they have no value

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/OffenderAlertRepository.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderAlert
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderAlertId
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderBooking
+
+@Repository
+interface OffenderAlertRepository : JpaRepository<OffenderAlert, OffenderAlertId> {
+  @Query("select coalesce(max(id.sequence), 0) + 1 from OffenderAlert where id.offenderBooking = :offenderBooking")
+  fun getNextSequence(offenderBooking: OffenderBooking): Int
+}
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/OffenderAlertRepository.kt:14]: not used yet, but will do soon

File: src/main/resources/db/migration/data/R__2_3_1__REFERENCES_CODES_alerts.sql [hunk 3]
```diff
@@ -0,0 +1,102 @@
+INSERT INTO REFERENCE_CODES (DOMAIN, CODE, PARENT_DOMAIN, PARENT_CODE, DESCRIPTION, LIST_SEQ, ACTIVE_FLAG, SYSTEM_DATA_FLAG, EXPIRED_DATE) VALUES ('ALERT', 'A', null, null, 'Social Care', 12, 'Y', 'N', null);
```
Comment (andymarke) [inline src/main/resources/db/migration/data/R__2_3_1__REFERENCES_CODES_alerts.sql:1]: trying this out by grouping reference data in different files so they get so big

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderAlert.kt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  var modifyUserId: String? = null
+
+  @Column(name = "MODIFY_DATETIME", insertable = false, updatable = false)
+  var modifyDatetime: LocalDateTime? = null
+
+  @Column(name = "CREATE_DATETIME", insertable = false, updatable = false)
+  lateinit var createDatetime: LocalDateTime
+
+  @Column(name = "CREATE_USER_ID", insertable = false, updatable = false)
+  @Generated
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderAlert.kt:106]: Should the other audit columns have this annotation too?

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderAlert.kt (hunk 4 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderAlert.kt:106]: I think all the others are nullable, does that make a difference? Actually I think you are right... actually no I don't think I need them on any of them .. actually I DON'T KNOW!!!!!

**Pull request merged**